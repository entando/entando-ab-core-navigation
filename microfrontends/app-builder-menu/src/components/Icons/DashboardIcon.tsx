import { COLORS } from '../theme';

interface Props {
  width?: number;
  height?: number;
  fill?: string;
}

function DashboardIcon(props: Props): JSX.Element {
  const { width = 20, height = 20, fill = COLORS.inactive } = props;
  return (
    <svg
      width={width}
      height={height}
      viewBox={`0 0 ${width} ${height}`}
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        fillRule="evenodd"
        clipRule="evenodd"
        d="M1.875 1.25H18.125C19.1602 1.25 20 2.08984 20 3.125V16.875C20 17.9102 19.1602 18.75 18.125 18.75H1.875C0.839844 18.75 0 17.9102 0 16.875V3.125C0 2.08984 0.839844 1.25 1.875 1.25ZM17.8906 16.875C18.0195 16.875 18.125 16.7695 18.125 16.6406V7.5H1.875V16.6406C1.875 16.7695 1.98047 16.875 2.10938 16.875H17.8906Z"
        fill="black"
      />
      <mask
        id="mask0_277_1709"
        maskUnits="userSpaceOnUse"
        x="0"
        y="1"
        width="20"
        height="18"
      >
        <path
          fillRule="evenodd"
          clipRule="evenodd"
          d="M1.875 1.25H18.125C19.1602 1.25 20 2.08984 20 3.125V16.875C20 17.9102 19.1602 18.75 18.125 18.75H1.875C0.839844 18.75 0 17.9102 0 16.875V3.125C0 2.08984 0.839844 1.25 1.875 1.25ZM17.8906 16.875C18.0195 16.875 18.125 16.7695 18.125 16.6406V7.5H1.875V16.6406C1.875 16.7695 1.98047 16.875 2.10938 16.875H17.8906Z"
          fill="white"
        />
      </mask>
      <g mask="url(#mask0_277_1709)">
        <path
          fillRule="evenodd"
          clipRule="evenodd"
          d="M0 0H20V20H0V0Z"
          fill={fill}
        />
      </g>
    </svg>
  );
}

export default DashboardIcon;
