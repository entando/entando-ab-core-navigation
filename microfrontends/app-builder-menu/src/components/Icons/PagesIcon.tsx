import { COLORS } from '../theme';

interface Props {
  width?: number;
  height?: number;
  fill?: string;
}

function PagesIcon(props: Props): JSX.Element {
  const { width = 20, height = 20, fill = COLORS.inactive } = props;
  return (
    <svg
      width={width}
      height={height}
      viewBox={`0 0 ${width} ${height}`}
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        fillRule="evenodd"
        clipRule="evenodd"
        d="M16.1742 0.54918L18.2008 2.57582C18.5525 2.92745 18.75 3.40436 18.75 3.90164V14.375C18.75 15.4105 17.9105 16.25 16.875 16.25H13.75V18.125C13.75 19.1605 12.9105 20 11.875 20H3.125C2.08945 20 1.25 19.1605 1.25 18.125V5.625C1.25 4.58945 2.08945 3.75 3.125 3.75H6.25V1.875C6.25 0.839453 7.08945 0 8.125 0H14.8484C15.3456 2.6003e-06 15.8226 0.197548 16.1742 0.54918ZM3.35938 18.125H11.6406C11.7701 18.125 11.875 18.0201 11.875 17.8906V16.25H8.125C7.08945 16.25 6.25 15.4105 6.25 14.375V5.625H3.35938C3.22993 5.625 3.125 5.72993 3.125 5.85938V17.8906C3.125 18.0201 3.22993 18.125 3.35938 18.125ZM8.35938 14.375C8.22993 14.375 8.125 14.2701 8.125 14.1406V2.10938C8.125 1.97993 8.22993 1.875 8.35938 1.875H12.5V5.3125C12.5 5.83027 12.9197 6.25 13.4375 6.25H16.875V14.1406C16.875 14.2701 16.7701 14.375 16.6406 14.375H8.35938ZM14.375 4.375H16.875V3.99875C16.875 3.93659 16.8503 3.87697 16.8064 3.83301L14.917 1.94363C14.873 1.89969 14.8134 1.875 14.7513 1.875H14.375V4.375Z"
        fill="black"
      />
      <mask
        id="mask0_277_1596"
        maskUnits="userSpaceOnUse"
        x="1"
        y="0"
        width="18"
        height="20"
      >
        <path
          fillRule="evenodd"
          clipRule="evenodd"
          d="M16.1742 0.54918L18.2008 2.57582C18.5525 2.92745 18.75 3.40436 18.75 3.90164V14.375C18.75 15.4105 17.9105 16.25 16.875 16.25H13.75V18.125C13.75 19.1605 12.9105 20 11.875 20H3.125C2.08945 20 1.25 19.1605 1.25 18.125V5.625C1.25 4.58945 2.08945 3.75 3.125 3.75H6.25V1.875C6.25 0.839453 7.08945 0 8.125 0H14.8484C15.3456 2.6003e-06 15.8226 0.197548 16.1742 0.54918ZM3.35938 18.125H11.6406C11.7701 18.125 11.875 18.0201 11.875 17.8906V16.25H8.125C7.08945 16.25 6.25 15.4105 6.25 14.375V5.625H3.35938C3.22993 5.625 3.125 5.72993 3.125 5.85938V17.8906C3.125 18.0201 3.22993 18.125 3.35938 18.125ZM8.35938 14.375C8.22993 14.375 8.125 14.2701 8.125 14.1406V2.10938C8.125 1.97993 8.22993 1.875 8.35938 1.875H12.5V5.3125C12.5 5.83027 12.9197 6.25 13.4375 6.25H16.875V14.1406C16.875 14.2701 16.7701 14.375 16.6406 14.375H8.35938ZM14.375 4.375H16.875V3.99875C16.875 3.93659 16.8503 3.87697 16.8064 3.83301L14.917 1.94363C14.873 1.89969 14.8134 1.875 14.7513 1.875H14.375V4.375Z"
          fill="white"
        />
      </mask>
      <g mask="url(#mask0_277_1596)">
        <path
          fillRule="evenodd"
          clipRule="evenodd"
          d="M0 0H20V20H0V0Z"
          fill={fill}
        />
      </g>
    </svg>
  );
}

export default PagesIcon;
