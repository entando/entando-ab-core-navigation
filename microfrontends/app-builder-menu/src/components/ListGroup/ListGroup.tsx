import styled from 'styled-components';

export const ListGroup = styled.ul`
  border-top: 0;
  margin-bottom: 0;
`;
