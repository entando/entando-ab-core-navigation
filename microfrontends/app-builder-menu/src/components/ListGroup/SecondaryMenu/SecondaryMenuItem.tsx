import styled from 'styled-components';
import RightArrowIcon from '../../Icons/RightArrowIcon';
import TertiaryMenu from './TertiaryMenu';

const StyledLink = styled.a<{ isActive: boolean }>`
  width: 100%;
  background-color: transparent;
  background-color: ${({ isActive }) => (isActive ? '#4D5257' : 'transparent')};
  color: ${({ isActive }) => (isActive ? '#fff' : '#d1d1d1')};
  cursor: pointer;
  display: block;
  font-size: 14px;
  font-weight: ${({ isActive }) => (isActive ? '600' : '400')};
  line-height: 26px;
  padding: 5px 10px 5px 8px;
  position: relative;
  white-space: nowrap;
  text-decoration: none;
  display: flex;
  margin-top: 4px;

  &:hover {
    font-weight: 600;
    color: #fff;
    background-color: #4d5257;
  }
`;

const StyledLabel = styled.span`
  display: block;
  flex: 1;
  max-width: none;
  overflow: hidden;
  text-overflow: ellipsis;
`;

const StyledSecondaryMenuItem = styled.li`
  display: flex;
`;

interface Props {
  isActive: boolean;
  hasChildren: boolean;
}

export default function SecondaryMenuItem(props: Props): JSX.Element {
  const { isActive, hasChildren } = props;
  return (
    <StyledSecondaryMenuItem>
      <StyledLink isActive={isActive}>
        <StyledLabel>Dashboard</StyledLabel>
        {hasChildren && <RightArrowIcon />}
      </StyledLink>
      <TertiaryMenu isOpen />
    </StyledSecondaryMenuItem>
  );
}
