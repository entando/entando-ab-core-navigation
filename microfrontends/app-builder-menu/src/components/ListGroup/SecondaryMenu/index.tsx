import { useContext } from 'react';
import styled from 'styled-components';
import BackButtonIcon from '../../Icons/BackButtonIcon';
import MenuUIContext from '../../MenuUIContext';
import { COLORS } from '../../theme';

const StyledSecondaryNavMenu = styled.div<{ isVisible: boolean }>`
  background: ${COLORS.active};
  border: 1px solid ${COLORS.primary};
  border-bottom: none;
  border-top: none;
  bottom: 0;
  display: block;
  left: 200px;
  overflow-x: hidden;
  overflow-y: auto;
  position: fixed;
  top: 0px;
  opacity: 0;
  visibility: hidden;
  ${({ isVisible }) => (isVisible ? `opacity: 1; visibility: visible;` : ``)}
  width: 200px;
  z-index: 1030;
`;

const StyledHeader = styled.div`
  display: flex;
  align-items: center;
  color: ${COLORS.text};
  font-weight: 600;
  margin-top: 19px;
  margin-left: 28px;
`;

const StyledTitle = styled.span`
  margin-left: 14px;
`;

const StyledSecondaryListGroup = styled.ul`
  margin-top: 23px;
  padding-left: 24px;
  padding-right: 0px;
`;

const StyledBackButton = styled.div`
  display: flex;
  align-items: center;
  cursor: pointer;
`;

interface Props {
  isOpen: boolean;
  title: string;
  children?: React.ReactNode;
}

export default function SecondaryMenu(props: Props): JSX.Element {
  const { isOpen, title, children } = props;

  const { setSecondaryMenuOpen, setTertiaryMenuOpen } =
    useContext(MenuUIContext);

  const handleBack = () => {
    setSecondaryMenuOpen(false);
    setTertiaryMenuOpen(false);
  };

  return (
    <StyledSecondaryNavMenu isVisible={isOpen}>
      <StyledHeader>
        <StyledBackButton onClick={handleBack}>
          <BackButtonIcon />
        </StyledBackButton>
        <StyledTitle>{title}</StyledTitle>
      </StyledHeader>
      <StyledSecondaryListGroup>{children}</StyledSecondaryListGroup>
    </StyledSecondaryNavMenu>
  );
}
