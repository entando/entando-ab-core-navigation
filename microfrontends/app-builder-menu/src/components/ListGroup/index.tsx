import styled from 'styled-components';

const ListGroup = styled.ul`
  border-top: 0;
  margin-bottom: 0;
`;

export default ListGroup;
