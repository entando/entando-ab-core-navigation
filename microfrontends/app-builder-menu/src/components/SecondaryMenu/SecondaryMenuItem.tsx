import { SyntheticEvent, useContext } from 'react';
import styled from 'styled-components';
import { RightArrowIcon } from './../Icons/RightArrowIcon';
import { MenuUIContext } from './../MenuUIContext';
import { TertiaryMenu } from '../TertiaryMenu/TertiaryMenu';
import { COLORS } from './../theme';

const StyledLink = styled.a<{ isActive: boolean }>`
  width: 100%;
  background-color: transparent;
  background-color: ${({ isActive }) =>
    isActive ? COLORS.secondary : 'transparent'};
  color: ${({ isActive }) => (isActive ? COLORS.white : COLORS.text)};
  cursor: pointer;
  display: block;
  font-size: 14px;
  font-family: 'Open Sans', sans-serif;
  font-weight: ${({ isActive }) => (isActive ? '600' : '400')};
  line-height: 26px;
  padding: 5px 10px 5px 8px;
  position: relative;
  white-space: nowrap;
  text-decoration: none;
  display: flex;
  margin-top: 4px;

  &:hover {
    font-weight: 600;
    color: ${COLORS.white};
    background-color: ${COLORS.secondary};
  }
`;

const StyledLabel = styled.span`
  display: block;
  flex: 1;
  max-width: none;
  overflow: hidden;
  text-overflow: ellipsis;
`;

const StyledSecondaryMenuItem = styled.li`
  display: flex;
`;

interface Props {
  id: string;
  dataId: string;
  children?: React.ReactNode;
  label: string;
  onClick?: () => void;
  href?: string;
  className?: string;
}

export function SecondaryMenuItem(props: Props): JSX.Element {
  const {
    id,
    dataId,
    children,
    label,
    onClick,
    href,
    className = '',
  } = props;

  const {
    activeSecondaryMenuItemId,
    setActiveSecondaryMenuItemId,
    tertiaryMenuOpen,
    setTertiaryMenuOpen
  } = useContext(MenuUIContext);

  const hasChildren = !!children;
  const isActive = activeSecondaryMenuItemId === id;

  const handleClick = (e: SyntheticEvent): void => {
    e.stopPropagation()

    setActiveSecondaryMenuItemId(id);
    if (hasChildren) {
      setTertiaryMenuOpen(true);
    } else {
      onClick && onClick();
    }
  };

  return (
    <>
      <StyledSecondaryMenuItem
        onClick={handleClick}
        className={className}
        data-id={dataId}
      >
        <StyledLink isActive={isActive} href={href}>
          <StyledLabel>{label}</StyledLabel>
          {hasChildren && <RightArrowIcon />}
        </StyledLink>
        {hasChildren && isActive && tertiaryMenuOpen && (
          <TertiaryMenu isOpen>{children}</TertiaryMenu>
        )}
      </StyledSecondaryMenuItem>
    </>
  );
}
