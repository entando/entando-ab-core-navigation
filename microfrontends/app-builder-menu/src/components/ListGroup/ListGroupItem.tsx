import { useContext } from 'react';
import styled from 'styled-components';
import { RightArrowIcon } from '../Icons/RightArrowIcon';
import { MenuUIContext } from '../MenuUIContext';
import { SecondaryMenu } from '../SecondaryMenu/SecondaryMenu';
import { COLORS } from '../theme';

interface ListGroupItemProps {
  fixBottom?: boolean;
  isActive?: boolean;
}

const StyledListGroupItem = styled.li<ListGroupItemProps>`
  position: relative;
  display: block;
  padding: 10px 15px;
  margin-bottom: -1px;
  border: 1px solid ${COLORS.border};
  border-left: 0;
  border-right: 0;
  background-color: transparent;
  padding: 0;
  color: ${({ isActive }) => (isActive ? COLORS.white : '')};
  ${({ fixBottom }) =>
    fixBottom
      ? `
  position: fixed;
  bottom: 64px;
  left: 0;
  `
      : ''}

  &:first-child {
    border-top: 0px;
  }
`;

interface StyledLinkProps {
  isActive: boolean;
}

const StyledLink = styled.a<StyledLinkProps>`
  width: 215px;
  background-color: ${({ isActive }) =>
    isActive ? COLORS.active : 'transparent'};
  color: ${({ isActive }) => (isActive ? COLORS.white : COLORS.text)};
  cursor: pointer;
  display: block;
  font-size: 14px;
  font-family: 'Open Sans', sans-serif;
  font-weight: ${({ isActive }) => (isActive ? '600' : '400')};
  height: 63px;
  line-height: 26px;
  padding: 17px 20px 17px 25px;
  position: relative;
  white-space: nowrap;
  text-decoration: none;
  display: flex;
  padding-right: 0;
  ${({ isActive }) =>
    isActive
      ? `
  &:before {
    background: ${COLORS.blue};
    content: " ";
    height: 100%;
    left: 0;
    position: absolute;
    top: 0;
    width: 3px;
  }
  `
      : ''}

  &:hover {
    font-weight: 600;
    color: ${COLORS.white};
    background-color: ${COLORS.active};
    &:before {
      background: ${COLORS.blue};
      content: ' ';
      height: 100%;
      left: 0;
      position: absolute;
      top: 0;
      width: 3px;
    }

    span > svg > path {
      fill: ${COLORS.blue};
    }
    span > svg > g > path {
      fill: ${COLORS.blue};
    }
  }
`;

const StyledIcon = styled.span`
  display: inline-flex;
  align-items: center;
  color: ${COLORS.inactive};
  float: left;
  font-size: 20px;
  line-height: 26px;
  margin-right: 10px;
  text-align: center;
  width: 24px;
`;

export const StyledLabel = styled.span`
  display: flex;
  line-height: 25px;
  flex: 1;
  max-width: none;
  padding-right: 15px;
  overflow: hidden;
  text-overflow: ellipsis;
  align-items: center;
`;

interface Props {
  id: string;
  fixBottom?: boolean;
  label: string;
  renderIcon: ({ fill }: { fill?: string }) => React.ReactNode;
  children?: React.ReactNode;
  onClick?: () => void;
  href?: string;
  className?: string;
}

export function ListGroupItem(props: Props): JSX.Element {
  const {
    label,
    renderIcon,
    fixBottom,
    children,
    id,
    onClick,
    href,
    className = ''
  } = props;

  const {
    activeListGroupItemId,
    setActiveListGroupItemId,
    setSecondaryMenuOpen,
    secondaryMenuOpen,
    setActiveTertiaryMenuItemId,
    setActiveSecondaryMenuItemId
  } = useContext(MenuUIContext);

  const isActive = activeListGroupItemId === id;
  const hasChildren = !!children;

  const onClickHandler = () => {
    setActiveListGroupItemId(id);
    if (hasChildren) {
      setSecondaryMenuOpen(true);
      setActiveTertiaryMenuItemId('');
      setActiveSecondaryMenuItemId('');
    } else {
      onClick && onClick();
    }
  };

  return (
    <>
      <StyledListGroupItem
        isActive={isActive}
        fixBottom={fixBottom}
        onClick={onClickHandler}
        className={className}
      >
        <StyledLink isActive={isActive} href={href}>
          <StyledIcon>
            {renderIcon({ fill: isActive ? COLORS.blue : COLORS.inactive })}
          </StyledIcon>
          <StyledLabel>{label}</StyledLabel>
          {hasChildren && (
            <StyledIcon>
              <RightArrowIcon />{' '}
            </StyledIcon>
          )}
        </StyledLink>
      </StyledListGroupItem>
      {hasChildren && isActive && secondaryMenuOpen && (
        <SecondaryMenu isOpen title={label}>
          {children}
        </SecondaryMenu>
      )}
    </>
  );
}
