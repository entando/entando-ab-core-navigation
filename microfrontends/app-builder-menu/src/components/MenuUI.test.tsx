import React from 'react';
import renderer from 'react-test-renderer';
import { DashboardIcon } from './Icons/DashboardIcon';
import { PagesIcon } from './Icons/PagesIcon';

import { ListGroup } from './ListGroup/ListGroup';
import { ListGroupItem } from './ListGroup/ListGroupItem';
import { SecondaryMenuItem } from './SecondaryMenu/SecondaryMenuItem';

test('renders MenuUI', () => {
  const tree = renderer
    .create(
      <ListGroup>
        <ListGroupItem
          id="dashboard"
          dataId="dashboard"
          label="Dashboard"
          renderIcon={props => <DashboardIcon {...props} />}
        />
        <ListGroupItem
          id="pages"
          dataId="pages"
          label="Pages"
          className="app-tour-step-3"
          renderIcon={props => <PagesIcon {...props} />}
        >
          <SecondaryMenuItem
            id="pages-management"
            dataId="pages-management"
            label="Management"
            className="app-tour-step-4"
          />
          <SecondaryMenuItem id="pages-designer" dataId="pages-designer" label="Designer" />
          <SecondaryMenuItem id="pages-templates" dataId="pages-templates" label="Templates" />
          <SecondaryMenuItem id="pages-settings" dataId="pages-settings" label="Settings" />
        </ListGroupItem>
      </ListGroup>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
