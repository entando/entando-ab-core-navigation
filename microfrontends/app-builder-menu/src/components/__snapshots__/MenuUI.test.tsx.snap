// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MenuUI 1`] = `
<ul
  className="sc-bczRLJ AKbNS"
>
  <li
    className="sc-gKXOVf iJxYWw"
    onClick={[Function]}
  >
    <a
      className="sc-iBkjds ekYlN"
    >
      <span
        className="sc-ftvSup chxByE"
      >
        <svg
          fill="none"
          height={20}
          viewBox="0 0 20 20"
          width={20}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M1.875 1.25H18.125C19.1602 1.25 20 2.08984 20 3.125V16.875C20 17.9102 19.1602 18.75 18.125 18.75H1.875C0.839844 18.75 0 17.9102 0 16.875V3.125C0 2.08984 0.839844 1.25 1.875 1.25ZM17.8906 16.875C18.0195 16.875 18.125 16.7695 18.125 16.6406V7.5H1.875V16.6406C1.875 16.7695 1.98047 16.875 2.10938 16.875H17.8906Z"
            fill="black"
            fillRule="evenodd"
          />
          <mask
            height="18"
            id="mask0_277_1709"
            maskUnits="userSpaceOnUse"
            width="20"
            x="0"
            y="1"
          >
            <path
              clipRule="evenodd"
              d="M1.875 1.25H18.125C19.1602 1.25 20 2.08984 20 3.125V16.875C20 17.9102 19.1602 18.75 18.125 18.75H1.875C0.839844 18.75 0 17.9102 0 16.875V3.125C0 2.08984 0.839844 1.25 1.875 1.25ZM17.8906 16.875C18.0195 16.875 18.125 16.7695 18.125 16.6406V7.5H1.875V16.6406C1.875 16.7695 1.98047 16.875 2.10938 16.875H17.8906Z"
              fill="white"
              fillRule="evenodd"
            />
          </mask>
          <g
            mask="url(#mask0_277_1709)"
          >
            <path
              clipRule="evenodd"
              d="M0 0H20V20H0V0Z"
              fill="#72767b"
              fillRule="evenodd"
            />
          </g>
        </svg>
      </span>
      <span
        className="sc-papXJ hMtRuD"
      >
        Dashboard
      </span>
    </a>
  </li>
  <li
    className="sc-gKXOVf iJxYWw app-tour-step-3"
    onClick={[Function]}
  >
    <a
      className="sc-iBkjds ekYlN"
    >
      <span
        className="sc-ftvSup chxByE"
      >
        <svg
          fill="none"
          height={20}
          viewBox="0 0 20 20"
          width={20}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M16.1742 0.54918L18.2008 2.57582C18.5525 2.92745 18.75 3.40436 18.75 3.90164V14.375C18.75 15.4105 17.9105 16.25 16.875 16.25H13.75V18.125C13.75 19.1605 12.9105 20 11.875 20H3.125C2.08945 20 1.25 19.1605 1.25 18.125V5.625C1.25 4.58945 2.08945 3.75 3.125 3.75H6.25V1.875C6.25 0.839453 7.08945 0 8.125 0H14.8484C15.3456 2.6003e-06 15.8226 0.197548 16.1742 0.54918ZM3.35938 18.125H11.6406C11.7701 18.125 11.875 18.0201 11.875 17.8906V16.25H8.125C7.08945 16.25 6.25 15.4105 6.25 14.375V5.625H3.35938C3.22993 5.625 3.125 5.72993 3.125 5.85938V17.8906C3.125 18.0201 3.22993 18.125 3.35938 18.125ZM8.35938 14.375C8.22993 14.375 8.125 14.2701 8.125 14.1406V2.10938C8.125 1.97993 8.22993 1.875 8.35938 1.875H12.5V5.3125C12.5 5.83027 12.9197 6.25 13.4375 6.25H16.875V14.1406C16.875 14.2701 16.7701 14.375 16.6406 14.375H8.35938ZM14.375 4.375H16.875V3.99875C16.875 3.93659 16.8503 3.87697 16.8064 3.83301L14.917 1.94363C14.873 1.89969 14.8134 1.875 14.7513 1.875H14.375V4.375Z"
            fill="black"
            fillRule="evenodd"
          />
          <mask
            height="20"
            id="mask0_277_1596"
            maskUnits="userSpaceOnUse"
            width="18"
            x="1"
            y="0"
          >
            <path
              clipRule="evenodd"
              d="M16.1742 0.54918L18.2008 2.57582C18.5525 2.92745 18.75 3.40436 18.75 3.90164V14.375C18.75 15.4105 17.9105 16.25 16.875 16.25H13.75V18.125C13.75 19.1605 12.9105 20 11.875 20H3.125C2.08945 20 1.25 19.1605 1.25 18.125V5.625C1.25 4.58945 2.08945 3.75 3.125 3.75H6.25V1.875C6.25 0.839453 7.08945 0 8.125 0H14.8484C15.3456 2.6003e-06 15.8226 0.197548 16.1742 0.54918ZM3.35938 18.125H11.6406C11.7701 18.125 11.875 18.0201 11.875 17.8906V16.25H8.125C7.08945 16.25 6.25 15.4105 6.25 14.375V5.625H3.35938C3.22993 5.625 3.125 5.72993 3.125 5.85938V17.8906C3.125 18.0201 3.22993 18.125 3.35938 18.125ZM8.35938 14.375C8.22993 14.375 8.125 14.2701 8.125 14.1406V2.10938C8.125 1.97993 8.22993 1.875 8.35938 1.875H12.5V5.3125C12.5 5.83027 12.9197 6.25 13.4375 6.25H16.875V14.1406C16.875 14.2701 16.7701 14.375 16.6406 14.375H8.35938ZM14.375 4.375H16.875V3.99875C16.875 3.93659 16.8503 3.87697 16.8064 3.83301L14.917 1.94363C14.873 1.89969 14.8134 1.875 14.7513 1.875H14.375V4.375Z"
              fill="white"
              fillRule="evenodd"
            />
          </mask>
          <g
            mask="url(#mask0_277_1596)"
          >
            <path
              clipRule="evenodd"
              d="M0 0H20V20H0V0Z"
              fill="#72767b"
              fillRule="evenodd"
            />
          </g>
        </svg>
      </span>
      <span
        className="sc-papXJ hMtRuD"
      >
        Pages
      </span>
      <span
        className="sc-ftvSup chxByE"
      >
        <svg
          fill="none"
          height={20}
          stroke="#979797"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 20 20"
          width={20}
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="9 18 15 12 9 6"
          />
        </svg>
         
      </span>
    </a>
  </li>
</ul>
`;
